staging/src/k8s.io/apiserver/pkg/cel/environment/base.go

```diff
@@ -81,7 +87,12 @@ var baseOpts = []VersionedOptions{
 			library.Quantity(),
 		},
 	},
-	// TODO: switch to ext.Strings version 2 once format() is fixed to work with HomogeneousAggregateLiterals.
+	{
```

Comment:
nit: would it be easier to make sure we don't have gaps/holes/overlap to group the versions of the string library together with a comment, e.g. 

```go
  // string library
	{
		IntroducedVersion: version.MajorMinor(1, 0),
		RemovedVersion:    version.MajorMinor(1, 29),
		EnvOptions: []cel.EnvOption{
			ext.Strings(ext.StringsVersion(0)),
		},
	},
	{
		IntroducedVersion: version.MajorMinor(1, 29),
		EnvOptions: []cel.EnvOption{
			ext.Strings(ext.StringsVersion(2)),
		},
	},

```

also, I can't remember if we have automatic sanity checking that these ranges don't overlap or leave holes or put and end version and make functions unavailable

Reply:
Grouping makes sense. I'll add a test that checks that range coverage.

Reply:
Done. The added unit test lints for a few things including version overlap and gaps.